### PR TITLE
Use nodemailer's callback style

### PR DESCRIPTION
**Pull request by elbuo8**

### CODE REVIEW

File: lib/mandrill-transport.js [hunk 1]
```diff
@@ -37,16 +37,16 @@ MandrillTransport.prototype.send = function(mail, callback) {
...
-    }
+  }, function(results) {
+
+    var info = {accepted: [], rejected: []};
+    results.forEach(function(result) {
+      if (['sent', 'queued', 'scheduled'].indexOf(result.status) !== -1)
+        info.accepted.push(result);
+      else
+        info.rejected.push(result);
+    });
```
Comment (motdotla) [inline lib/mandrill-transport.js]: This looks right, but I'm looking into Mandrill's responses in more detail.

But, I think we should be using braces with multiline blocks - https://github.com/RebelMail/javascript#blocks

Comment (motdotla) [inline lib/mandrill-transport.js]: Ok, those looks right - 'rejected', and 'invalid' into info.rejected - cool.

Comment (elbuo8) [inline lib/mandrill-transport.js]: @motdotla added.

**Comment by motdotla** [comment]
This should get @devinus review too, but :+1:

**Comment by devinus** [comment]
This loses `messageId`?

**Comment by elbuo8** [comment]
Yes. Mandrill uses *per message* id. Setting a higher level one would just be ignoring/overriding the rest of them.

**Comment by elbuo8** [comment]
We could, return an array of `messageId`'s as an alternative.

**Comment by motdotla** [comment]
This is tricky. It should be a string since that is expected always that I can tell.

It looks like a random one is generated by default though if missing.

> `messageId` - optional Message-Id value, random value will be generated if not set

How about we just let the random value be generated?

[1] https://github.com/andris9/Nodemailer#e-mail-message-fields

**Comment by elbuo8** [comment]
@motdotla random is for sending for what I can observe.

**Comment by devinus** [comment]
This is the default `messageId` implementation used when one isn't found in the headers:

https://github.com/andris9/buildmail/blob/3d4a38ec1d8ac1d5a9f58d9bbbc446756259ed70/src/buildmail.js#L427

**Comment by motdotla** [comment]
Good find @devinus. I think that is probably the best solution then - just let it autogenerate. Agree, disagree, guys?

**Comment by elbuo8** [comment]
So basically this will be passed down in the mail object?

**Comment by motdotla** [comment]
Yep

**Comment by elbuo8** [comment]
Added. @motdotla @devinus

**Comment by devinus** [comment]
I'm not sure if it's part of `mail.data`. In fact, I think it might be part of `mail.message`

**Comment by devinus** [comment]
Actually, I'm not sure if it's sent to at all by default. I think it's only generated for SMTP transport. Could easily test by inspecting data when sending with Nodemailer.

**Comment by motdotla** [comment]
Here's the object that comes back:

```
{ data: 
   { from: 'devinus@rebelmail.com',
     to: 'mot@rebelmail.com',
     subject: 'Hello',
     html: '<p>How are you?</p>',
     headers: {} },
  message: 
   { nodeCounter: 1,
     baseBoundary: '14322463038340.754273205762729',
     date: Thu May 21 2015 15:11:43 GMT-0700 (PDT),
     rootNode: [Circular],
     keepBcc: false,
     parentNode: undefined,
     childNodes: [],
     _nodeId: 1,
     _headers: 
      [ { key: 'Content-Type', value: 'text/html' },
        { key: 'From', value: 'devinus@rebelmail.com' },
        { key: 'To', value: 'mot@rebelmail.com' },
        { key: 'Subject', value: 'Hello' },
        { key: 'X-Mailer',
          value: 'nodemailer (1.3.4; +http://www.nodemailer.com; Mandrill/0.1.1)' } ],
     _isPlainText: true,
     _canUseFlowedContent: false,
     _isFlowedContent: false,
     _envelope: false,
     _transforms: [],
     content: '<p>How are you?</p>' },
  resolveContent: [Function] }
```

**Comment by motdotla** [comment]
So, messageId is NOT generated by default - that documentation was referencing it's own internal transports like SMTP transport.

Given that I would be for just using the first _id of the result response. That object looks like this on a multiple `to` send.

```
{ accepted: 
   [ { email: 'mot@rebelmail.com',
       status: 'sent',
       _id: '9e7dffd4fe704e66a2c45d48c02b8a7b',
       reject_reason: null },
     { email: 'mot@mot.la',
       status: 'sent',
       _id: '4bc281398c7e499daba122b43ff7d278',
       reject_reason: null } ],
  rejected: [],
  messageId: undefined }
```

**Comment by elbuo8** [comment]
:+1:

**Pull request merged**